### PR TITLE
[ErrorHandler] fix rendering exception pages without the HttpKernel component

### LINKED ISSUE #53430: [HttpKernel][ErrorHandler] Undocumented BC with logging
### Symfony version(s) affected

6.4

### Description

https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/ErrorHandler/CHANGELOG.md
nor 
https://github.com/symfony/symfony/blob/6.4/CHANGELOG-6.4.md

do not document the Breaking Change from 6.3 in [Symfony\Component\ErrorHandler\ErrorRenderer\HtmlErrorRenderer](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/ErrorHandler/ErrorRenderer/HtmlErrorRenderer.php)
for the now required [Symfony\Component\HttpKernel\Log\DebugLoggerConfigurator](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Log/DebugLoggerConfigurator) on [line 143](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/ErrorHandler/ErrorRenderer/HtmlErrorRenderer.php#L143)

### How to reproduce

```
composer init
composer require symfony/error-handler
```
Create an index.php file with the contents:
```
<?php

use Symfony\Component\ErrorHandler\BufferingLogger;
use Symfony\Component\ErrorHandler\DebugClassLoader;
use Symfony\Component\ErrorHandler\ErrorHandler;

require __DIR__ . '/vendor/autoload.php';

ErrorHandler::register(new ErrorHandler(new BufferingLogger(), true));

throw new \RuntimeException();
```
Run `php -S localhost:8000 index.php` and navigate to `localhost:8000`

Result:
<b>Fatal error</b>:  Uncaught Symfony\Component\ErrorHandler\Error\ClassNotFoundError: Attempted to load class &quot;DebugLoggerConfigurator&quot; from namespace &quot;Symfony\Component\HttpKernel\Log&quot;.
Did you forget a &quot;use&quot; statement for another namespace? in D:\php\symfony-error-handler-test\vendor\symfony\error-handler\ErrorRenderer\HtmlErrorRenderer.php:143
Stack trace:
#0 D:\php\symfony-error-handler-test\vendor\symfony\error-handler\ErrorRenderer\HtmlErrorRenderer.php(70): Symfony\Component\ErrorHandler\ErrorRenderer\HtmlErrorRenderer-&gt;renderException()
#1 D:\php\symfony-error-handler-test\vendor\symfony\error-handler\ErrorHandler.php(645): Symfony\Component\ErrorHandler\ErrorRenderer\HtmlErrorRenderer-&gt;render()
#2 D:\php\symfony-error-handler-test\vendor\symfony\error-handler\ErrorHandler.php(531): Symfony\Component\ErrorHandler\ErrorHandler-&gt;renderException()
#3 [internal function]: Symfony\Component\ErrorHandler\ErrorHandler-&gt;handleException()
#4 {main}
  thrown in <b>D:\php\symfony-error-handler-test\vendor\symfony\error-handler\ErrorRenderer\HtmlErrorRenderer.php</b> on line <b>143</b><br />

Expected result (without the requirement of having to `composer require symfony/http-kernel`):
image

### Possible Solution

Options:

1. Document the breaking change
2. Move the `Symfony\Component\HttpKernel\Log` classes into `symfony/error-handler` (`symfony/monolog-bridge` and `symfony/framework-bundle` and `symfony/http-kernel` both already require `symfony/error-handler`, see image) image
3. Revert this change

Option 2 "shouldn't" cause any issues, and will end up requiring less dependencies overall.

### Additional Context

none

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #53430
| License       | MIT

### CODE REVIEW

**Review by yceruto — APPROVED** [review]

**Review by radar3301 — COMMENTED** [review]
`$this->logger && class_exists`

File: src/Symfony/Component/ErrorHandler/ErrorRenderer/HtmlErrorRenderer.php [hunk 1]
```diff
@@ -140,7 +140,7 @@ private function renderException(FlattenException $exception, string $debugTempl
             'exceptionMessage' => $exceptionMessage,
             'statusText' => $statusText,
             'statusCode' => $statusCode,
-            'logger' => DebugLoggerConfigurator::getDebugLogger($this->logger),
+            'logger' => class_exists(DebugLoggerConfigurator::class) ? DebugLoggerConfigurator::getDebugLogger($this->logger) : null,
```
Comment (fabpot) [inline src/Symfony/Component/ErrorHandler/ErrorRenderer/HtmlErrorRenderer.php]: @radar3301 suggested a small optimization that makes sense to me:

```suggestion
            'logger' => ($this->logger && class_exists(DebugLoggerConfigurator::class)) ? DebugLoggerConfigurator::getDebugLogger($this->logger) : null,
```

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/ErrorHandler/ErrorRenderer/HtmlErrorRenderer.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/ErrorHandler/ErrorRenderer/HtmlErrorRenderer.php]: I added a check for the property not being `null`

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Comment by radar3301** [comment]
~~Just checking... Will this get merged automatically into 7.0 and 7.1?~~

Nevermind, I just saw that it did.

**Pull request merged**